src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -296,12 +300,48 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
                     services.TryAddEnumerable(ServiceDescriptor.Singleton<IHostedService, FunctionsScaleMonitorService>());
                 }
                 services.TryAddSingleton<FunctionsScaleManager>();
+
+                services.AddHostOverrides();
             });
 
             RegisterFileProvisioningService(builder);
             return builder;
         }
 
+        public static void AddHostOverrides(this IServiceCollection services)
+        {
+            // Custom implementations that the Host overrides
+            services.AddSingleton<ScheduleMonitor, AzureStorageScheduleMonitor>();
+            services.AddSingleton<IDistributedLockManager>(provider => AddLockManager(provider));
+        }
+
+        public static IDistributedLockManager AddLockManager(IServiceProvider provider)
+        {
+            try
+            {
+                var azureStorageProvider = provider.GetRequiredService<IAzureStorageProvider>();
+                var loggerFactory = provider.GetRequiredService<ILoggerFactory>();
+                var container = azureStorageProvider.GetBlobContainerClient();
```

Comment:
Not clear to me why we're creating this client here - it's not being used below? It's just a validation check? Looks like internally inside BlobLeaseDistributedLockManager you're doing the same call again. Why not just pass the client in on construction?

Reply:
This follows the similar logic in WebJobs to check if Blob storage is available before registering an implementation of IDistributedLockManager. Creating the client here gives us this validation to choose BlobLeaseDistributedLockManager; or otherwise just use In-Memory Lock.

It is only 1 additional GetContainerClient() call (which I don't think results in a network call) whenever the ScriptHost is recreated. To cut down on this call, I could save the reference in the GetContainerClient() if you think that'll help.

Reply:
Also, the method can be private since only used above?